sdks/python/apache_beam/coders/row_coder.py

```diff
@@ -87,7 +87,10 @@ def from_runner_api_parameter(schema, components, unused_context):
   @staticmethod
   def from_type_hint(type_hint, registry):
     if isinstance(type_hint, row_type.RowTypeConstraint):
-      schema = named_fields_to_schema(type_hint._fields)
+      try:
+        schema = named_fields_to_schema(type_hint._fields)
+      except ValueError:
+        return typecoders.registry.get_coder(object)
```

Comment:
Is this a fallback for when a Row uses types that we don't support in Python schemas?

I worry about this since it makes it tricky for a user to tell when a PCollection can be used in an ExternalTransform that uses rows. Are there some specific types that we need to add coverage for?

Reply:
Well, it's also there to handle the case where we simply don't know what the type is. I'll see if I can re-work this to use named tuples (thanks for the idea about `__reduce__`).

Reply:
Ah I see. I'd rather not have this escape hatch if we can avoid it, but I can imagine it would be frustrating for users who don't care about using Rows xlang if we just refuse to make a RowCoder for them. Perhaps we should make `named_fields_to_schema` wrap unknown type hints in a "pythonsdk" logical type that uses fast primitives coder. Then we can give a better error message if/when a xlang issue arises. If you just add a TODO here I can take that action.

When running locally I noticed that this is only needed for your tests because we can't infer a type for `sign=x // abs(x) if x else 0` and fallback to `Any`. Just curious - is it possible to fix the inference for that function? It seems like we should be able to.